ros2param/test/test_verb_dump.py

```diff
@@ -0,0 +1,131 @@
+        cls.node.declare_parameter('str_param', 'Hello World')
+        cls.node.declare_parameter('foo' + PARAMETER_SEPARATOR_STRING +
+                                   'str_param', 'foo')
+        cls.node.declare_parameter('foo' + PARAMETER_SEPARATOR_STRING +
+                                   'bar' + PARAMETER_SEPARATOR_STRING +
+                                   'str_param', 'foobar')
+
+        # We need both the test node and 'dump'
+        # node to be able to spin
+        cls.exec_thread = threading.Thread(target=cls.executor.spin)
+        cls.exec_thread.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.executor.shutdown()
+        cls.node.destroy_node()
+        rclpy.shutdown(context=cls.context)
+        cls.exec_thread.join()
+
+    def _output_file(self):
+
+        ns = self.node.get_namespace()
+        name = self.node.get_name()
+        if '/' == ns:
+            fqn = ns + name
+        else:
+            fqn = ns + '/' + name
+        return fqn[1:].replace('/', '__') + '.yaml'
+
+    def test_verb_dump_invalid_node(self):
+        assert cli.main(
+                argv=['param', 'dump', 'invalid_node']) == 'Node not found'
+
+        assert cli.main(
+                argv=['param', 'dump', 'invalid_ns/test_node']) == 'Node not found'
+
+    def test_verb_dump_invalid_path(self):
+        assert cli.main(
+                argv=['param', 'dump', 'foo/test_node', '--output-dir', 'invalid_path']) \
+                    == "'invalid_path' is not a valid directory."
+
+    def test_verb_dump(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            assert cli.main(
+                    argv=['param', 'dump', '/foo/test_node', '--output-dir', tmpdir]) is None
+
+            # Compare generated parameter file against expected
+            generated_param_file = os.path.join(tmpdir, self._output_file())
+            assert (open(generated_param_file, 'r').read() == EXPECTED_PARAMETER_FILE)
+
+    def test_verb_dump_print(self):
+        with patch('sys.stdout', new=StringIO()) as fake_stdout:
+            assert cli.main(
+                argv=['param', 'dump', 'foo/test_node', '--print']) is None
+
+            # Compare generated stdout against expected
+            assert fake_stdout.getvalue().strip() == EXPECTED_PARAMETER_FILE[:-1]
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            assert cli.main(
+                argv=['param', 'dump', 'foo/test_node', '--output-dir', tmpdir, '--print']) is None
+
+            not_generated_param_file = os.path.join(tmpdir, self._output_file())
+
+            with self.assertRaises(OSError) as context:
+                open(not_generated_param_file, 'r')
+
+            # Make sure the file was not create, thus '--print' did preempt
+            expected_err = '[Errno 2] No such file or directory: ' \
+                           "'{not_generated_param_file}'".format_map(locals())
```

Comment:
Looks like the error text is slightly different on windows. Maybe relax this check to `assert 'No such file or directory' in context.exception`?

https://ci.ros2.org/job/ci_windows/7522/testReport/junit/ros2param.test.test_verb_dump/TestVerbDump/test_verb_dump_print/

Reply:
That's surprising.  
See 7b508a0.